sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -514,12 +524,13 @@ private[sql] object Expand {
  * @param projections to apply
  * @param output of all projections.
  * @param child operator.
+ * @param groupByAttrs the attributes used in group by.
  */
 case class Expand(
     projections: Seq[Seq[Expression]],
     output: Seq[Attribute],
-    child: LogicalPlan) extends UnaryNode {
-
+    child: LogicalPlan,
+    groupByAttrs: Seq[Attribute]) extends UnaryNode {
```

Comment:
Thanks for pointing the problem. Agreed after re-thinking about it. As separating `projections` and `output` causes the problem. How about we get the output from `projections`?

As there are more than one projection, we can just get the output from the first projection and verify its consistency with other projections.